src/Shared/FileMatcher.cs

```diff
@@ -2570,7 +2584,7 @@ private static string[] CreateArrayWithSingleItemIfNotExcluded(string filespecUn
                 ? listOfFiles.SelectMany(list => list).Where(f => !resultsToExclude.Contains(f)).ToArray()
                 : listOfFiles.SelectMany(list => list).ToArray();
 
-            return files;
+            return (files, action);
```

Comment:
This line never gets reached due to an `UnauthorizedAccessException` that occurs inside of `GetFilesRecursive()` (see CreateItem_Tests and ProjectItem_Tests). Should the `SearchAction` `ReturnLogDriveEnumerationWildcard` force an early return in `GetFilesImplementation()`, and if so, what exactly should be returned?

Reply:
Note: Have to do manual testing to get a proper output for testing logged warnings